autode/wrappers/G09.py

```diff
@@ -36,180 +43,261 @@ def modify_keywords_for_point_charges(keywords):
     return None
 
 
-class G09(ElectronicStructureMethod):
-    # TODO implement partial hessian
+def get_keywords(calc_input, molecule):
+    """Modify the input keywords to try and fix some Gaussian's quirks"""
+
+    keywords = calc_input.keywords.copy()
+
+    # Mod redundant keywords is required if there are any constraints or
+    # modified internal coordinates
+    if molecule.constraints.any():
+        keywords.append('Geom=ModRedun')
```

Comment:
should the append method of a keyword have a check if the keyword is already present?

Reply:
Definitely, but need to be careful about giving keywords as lists at runtime

Reply:
could do assert type(keywords_set) == KeywordsSet in wrappers.base init

Reply:
Slightly more difficult to change the keywords at runtime but I think it;s worth it